libr/core/cbin.c

```diff
@@ -116,6 +116,10 @@ R_API int r_core_bin_set_env(RCore *r, RBinFile *binfile) {
 			r_config_set (r->config, "anal.cpu", arch);
 		}
 		r_asm_use (r->assembler, arch);
+		// reset all types information for r_core_anal_type_init
+		if (r->anal && r->anal->sdb_types) {
+			sdb_reset (r->anal->sdb_types);
```

Comment:
Why this? According to the description, it seems the problem is that there's a call to `sdb_reset`. Why the fix doesn't involve removing the culprit `sdb_reset`?

Reply:
I thought that maybe in some conditions we have to reset `sdb_types`. For example if user sets different arch we need to change standard types info.

In that case deletion of custom defined types is still undesirable. But this is more generic problem and I believe this is related to https://github.com/radare/radare2/issues/10996

Reply:
i think we should have more than one sdbtypes database. because user expects to keep the types loaded by him even after changing from one arch to another. but this involves some problematics like types changing in size between different archs and then having wrong user defined type definitions because not in sync